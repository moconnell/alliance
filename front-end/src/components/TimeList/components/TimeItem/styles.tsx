import styled from "styled-components";

export const TimeButton = styled.button`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 9px;
  font-size: 0.9em;
  font-weight: bold;
  border: 1px solid #ff3864;
  box-sizing: border-box;
  border-radius: 6px;
  color: #fff;
  margin-bottom: 10px;

  &:hover {
    opacity: 0.6;
    cursor: pointer;
  }
`;
