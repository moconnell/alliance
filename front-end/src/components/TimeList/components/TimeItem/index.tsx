import { DateTime } from "luxon";
import { TimeButton } from "./styles";

interface TimeItemProps {
  value: DateTime;
  active: boolean;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
}

const TimeItem = ({ value, active, onClick }: TimeItemProps) => (
  <TimeButton
    onClick={onClick}
    style={{ backgroundColor: active ? "#ff3864" : "transparent" }}
  >
    {value.toLocaleString(DateTime.TIME_SIMPLE)}
  </TimeButton>
);

export default TimeItem;
