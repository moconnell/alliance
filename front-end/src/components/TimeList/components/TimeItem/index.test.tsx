import { render } from "@testing-library/react";
import { DateTime } from "luxon";
import TimeItem from "./index";

describe("<TimeItem />", () => {
  const testData = [true, false];
  testData.forEach((active) =>
    it(`should render correctly when active=${active}`, async () => {
      const { asFragment } = render(
        <TimeItem active={active} value={DateTime.fromObject({ hour: 9, minute: 45 }).setLocale("en-us")} />
      );
      expect(asFragment()).toMatchSnapshot();
    })
  );
});
