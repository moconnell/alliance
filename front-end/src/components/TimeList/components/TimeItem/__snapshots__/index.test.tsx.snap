// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeItem /> should render correctly when active=false 1`] = `
<DocumentFragment>
  <button
    class="sc-bczRLJ fYCUyx"
    style="background-color: transparent;"
  >
    9:45 AM
  </button>
</DocumentFragment>
`;

exports[`<TimeItem /> should render correctly when active=true 1`] = `
<DocumentFragment>
  <button
    class="sc-bczRLJ fYCUyx"
    style="background-color: rgb(255, 56, 100);"
  >
    9:45 AM
  </button>
</DocumentFragment>
`;
